apps/emqx_plugins/src/emqx_plugins.erl

```diff
@@ -1148,7 +1181,7 @@ plugin_dir(NameVsn) ->
 
 -spec plugin_config_dir(name_vsn()) -> string().
 plugin_config_dir(NameVsn) ->
-    wrap_list_path(filename:join([plugin_dir(NameVsn), "data", "configs"])).
+    wrap_list_path(filename:join([emqx:data_dir(), "plugins", NameVsn])).
```

Comment:
Good idea to save to `data/plugins` dir.
Please ensure all config is deleted when uninstall/delete plugins.

Reply:
maybe we can not 🤔
Consider the situation of plugin upgrade:
The new version of the plugin still needs to use the configuration of the old version.

* Upgrade: The current implementation is that you must first delete the old version of the plug-in and then install the new version.

Reply:
1. Currently, I don't see anywhere deleting this `plugins/NameVersion/*`. I think it should be deleted when the user deletes the package.
2. Don't see any code do copy the old configuration to the new version when upgrading. And the new version of the configuration may not be compatible with the old version, just let the users handle it. **If the use already uninstall the package, it's mean all code and config should be deleted**, If we want to handle the upgrade, we should add a new action call **upgrade** in the future.

Reply:
The config file should not be deleted due to plugin-upgrade. Actual behavior is removal-installed plugin replacement process.
If the `plugins/Name/*` (note: **NO Version** as suffix) files be removed after plugin uninstalled.
The new version of plugin could not find the previous `config.hocon` file.